staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/typeconverter.go

```diff
@@ -35,25 +37,59 @@ type TypeConverter interface {
 	TypedToObject(*typed.TypedValue) (runtime.Object, error)
 }
 
+type typeResolver interface {
+	Type(schema.GroupVersionKind) *typed.ParseableType
+}
+
+type mapTypeResolver map[schema.GroupVersionKind]*typed.ParseableType
+
+func (m mapTypeResolver) Type(k schema.GroupVersionKind) *typed.ParseableType {
+	return m[k]
+}
+
 type typeConverter struct {
-	parser *managedfields.GvkParser
+	parser typeResolver
 }
 
 var _ TypeConverter = &typeConverter{}
 
-// NewTypeConverter builds a TypeConverter from a proto.Models. This
+// NewTypeConverter builds a TypeConverter from a spec.Swagger. This
 // will automatically find the proper version of the object, and the
 // corresponding schema information.
 func NewTypeConverter(openapiSpec *spec.Swagger, preserveUnknownFields bool) (TypeConverter, error) {
-	models, err := utilopenapi.ToProtoModels(openapiSpec)
-	if err != nil {
-		return nil, err
+	if openapiSpec == nil {
```

Comment:
That looks like a programming/set-up error rather than a transient error. Probably should panic.

Reply:
I think we should actually just remove the line since nil should be treated as empty